src/plugins/reporter-ureport.c

```diff
@@ -224,20 +232,23 @@ int main(int argc, char **argv)
         if (attach_value_from_rt)
         {
             report_result_t *result = find_in_reported_to(dd, report_result_type);
+            char *url;
 
             if (!result)
                 error_msg_and_die(_("This problem has not been reported to '%s'."), report_result_type);
 
+            url = report_result_get_url(result);
+
+            report_result_free(result);
+
             /* If you introduce a new attach_value_from_rt recognized value,
              * this condition will become invalid. */
-            if (!result->url)
+            if (!url)
                 error_msg_and_die(_("The report result '%s' is missing URL."), report_result_type);
 
             /* Avoid the need to duplicate the string. */
-            attach_value = attach_value_from_rt_data = result->url;
-            result->url = NULL;
-
-            free_report_result(result);
+            attach_value = url;
+            attach_value_from_rt_data = url;
```

Comment:
You should free the `url`.

Reply:
If you take a look at the `finalize` label, `attach_value_from_rt_data` is freed, which now holds `url`. It’s annoyingly complicated, but I didn’t notice a path that results in a leak.